net/usrsock/usrsock_recvfrom.c

```diff
@@ -418,6 +418,15 @@ ssize_t usrsock_recvfrom(FAR struct socket *psock, FAR void *buf, size_t len,
                */
 
               outaddrlen = state.valuelen_nontrunc;
+
+              /* If the MSG_PEEK flag is enabled, it will only peek
```

Comment:
But the data was consumed anyway? It's different from what MSG_PEEK specified.

Reply:
> But the data was consumed anyway? It's different from what MSG_PEEK specified.

@xiaoxiang781216 

I think it's up to daemon and driver implementation.
If they handle MSG_PEEK correctly, they will remove the packet from their queue.

Reply:
@xiaoxiang781216 

Sorry, this patch is depends on #1443. @masayuki2009's comment is correct.
For #1443, MSG_PEEK flag can be passed to daemon, and each daemon should be handle it correctly.

Reply:
So, we don't need this patch.

Reply:
No, this patch is needed. 
Actual buffer processing is in daemon, but usrsock wouldn't be passed the receive process because usrsock determines no data. So set this flag again for next recv() call.

Reply:
No, this patch is needed.
Actual buffer processing is in daemon, but usrsock wouldn't be passed the receive process because usrsock determines no data. So set this flag again for next recv() call.

Reply:
Ok.